packages/react-components/react-utilities-compat/library/src/memoize.ts

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable deprecation/deprecation */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Stylesheet } from './internals/Stylesheet';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+declare class WeakMap {
+  public get(key: any): any;
+  public set(key: any, value: any): void;
+  public has(key: any): boolean;
+}
+
+let _initializedStylesheetResets = false;
+let _resetCounter = 0;
+const _emptyObject = { empty: true };
+const _dictionary: any = {};
+let _weakMap = typeof WeakMap === 'undefined' ? null : WeakMap;
+
+interface IMemoizeNode {
+  map: WeakMap | null;
+  value?: any;
+}
+
+/**
+ *  @deprecated Consider using React.useMemo for components or your own update strategy for global values.
+ *
+ * Test utility for providing a custom weakmap.
+ *
+ * @internal
+ * */
+export function setMemoizeWeakMap(weakMap: any): void {
```

Comment:
why is this marked with `@internal` annotation when it's part of public API ?

Reply:
I don't know.  I brought this over from v8.